### PR TITLE
[TACHYON-1428] Documented the tachyon.security, tachyon.underfs and tachyon.util packages in preparation for the checkstyle rule

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1428

### CODE REVIEW

File: common/src/main/java/tachyon/security/login/AppLoginModule.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void initialize(Subject subject, CallbackHandler callbackHandler,
    * Retrieves the user name by querying the property of {@link Constants#SECURITY_LOGIN_USERNAME}.
    *
    * @return true if user name provided by application is set and not empty
-   * @throws javax.security.auth.login.LoginException
+   * @throws javax.security.auth.login.LoginException when the login fails
```
Comment (yupeng9) [inline common/src/main/java/tachyon/security/login/AppLoginModule.java]: Why does this exception use the full path?

File: common/src/main/java/tachyon/security/login/TachyonLoginModule.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public void initialize(Subject subject, CallbackHandler callbackHandler,
    *
    * The implementation does not really authenticate the user here. Always return true.
    * @return true in all cases
-   * @throws javax.security.auth.login.LoginException
+   * @throws javax.security.auth.login.LoginException when the login fails
```
Comment (yupeng9) [inline common/src/main/java/tachyon/security/login/TachyonLoginModule.java]: ditto

File: common/src/main/java/tachyon/security/login/AppLoginModule.java (hunk 1 above)
Comment (yupeng9) [inline common/src/main/java/tachyon/security/login/AppLoginModule.java]: I know this is from existing code, but could you help check it? Thanks!

File: common/src/main/java/tachyon/underfs/UnderFileSystemFactory.java [hunk 3]
```diff
@@ -23,14 +23,14 @@
 public interface UnderFileSystemFactory {
 
   /**
-   * Creates a new client for accessing the given path
+   * Creates a new client for accessing the given path. An {@link IllegalArgumentException} is
+   * thrown if this factory does not support clients for the given path or if the configuration
+   * provided is insufficient to create a client
```
Comment (yupeng9) [inline common/src/main/java/tachyon/underfs/UnderFileSystemFactory.java]: period at the end.

**Comment by yupeng9** [comment]
Thanks for the doc improvement! I left some comments.

**Comment by HorizonNet** [comment]
@yupeng9 Thanks for the comments. The update contains the corrections.

**Comment by yupeng9** [comment]
LGTM. Thanks.

**Pull request merged**